File: Python/HighFive.py
from typing import List

class Solution:
    def highFive(self, items: List[List[int]]) -> List[List[int]]:
        map = {}
        for i in items:
            if i[0] not in map:
                map[i[0]] = []
            map[i[0]].append(i[1])
        
        res = []

        print(map)
        for i in map:
            sum = 0
            l = sorted(map[i])
            
            for j in range(len(l)-5, len(l)):
                sum += l[j]
            
            res.append([i, sum//5])
        return res

test = Solution()
print(test.highFive([[1,91],[1,92],[2,93],[2,97],[1,60],[2,77],[1,65],[1,87],[1,100],[2,100],[2,76]]))

File: Java/src/HighFive.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;

public class HighFive {

    public int[][] highFive(int[][] items) {
        HashMap<Integer, ArrayList<Integer>> map = new HashMap<>();

        for(int[] i : items){
            if(!map.containsKey(i[0])) map.put(i[0], new ArrayList<>());
            map.get(i[0]).add(i[1]);
        }

        int[][] res = new int[map.size()][2];
        int idx = 0;
        for(int i : map.keySet()){
            ArrayList<Integer> l = map.get(i);
            Collections.sort(l);

            int sum = 0;
            for(int j = l.size()-1; j >= l.size()-5; j--) sum += l.get(j);

            res[idx][0] = i;
            res[idx++][1] = sum / 5;
        }

        return res;
    }

}